src/panels/lovelace/cards/hui-entity-button-card.ts

```diff
@@ -29,38 +35,46 @@ interface Config extends LovelaceConfig {
 class HuiEntityButtonCard extends hassLocalizeLitMixin(LitElement)
   implements LovelaceCard {
   public hass?: HomeAssistant;
-  protected config?: Config;
+  private _config?: Config;
 
   static get properties(): PropertyDeclarations {
     return {
       hass: {},
-      config: {},
+      _config: {},
     };
   }
 
-  public getCardSize() {
+  public getCardSize(): number {
     return 2;
   }
 
-  public setConfig(config: Config) {
+  public setConfig(config: Config): void {
     if (!isValidEntityId(config.entity)) {
       throw new Error("Invalid Entity");
     }
 
-    this.config = { theme: "default", ...config };
+    this._config = { theme: "default", ...config };
+  }
 
-    if (this.hass) {
-      this.requestUpdate();
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    const newHass = changedProps.get("hass") as any;
```

Comment:
```suggestion
    const newHass = changedProps.get("hass") as HomeAssistant | undefined;
```

Reply:
Why did you change this to oldHass?

Reply:
wait is changed props the old hass? I thought it was newer than hass hmm 🤔

Reply:
`this.hass` is the new one.